src/services/services.ts

```diff
@@ -2915,6 +3000,69 @@ namespace ts {
 
     export function createLanguageService(host: LanguageServiceHost,
         documentRegistry: DocumentRegistry = createDocumentRegistry(host.useCaseSensitiveFileNames && host.useCaseSensitiveFileNames(), host.getCurrentDirectory())): LanguageService {
+            const baseService = createUnextendedLanguageService(host, documentRegistry);
+            const extensions = baseService.getProgram().getCompilerExtensions()["language-service"];
+            const instantiatedExtensions = map(extensions, extension => new extension.ctor(ts, host, baseService, documentRegistry, extension.args));
+
+            function wrap(key: string): Function {
+                return (...args: any[]) => {
+                    if (instantiatedExtensions && instantiatedExtensions.length) {

```

Comment:
Consider moving this test outside the `wrap` call and never calling `wrap` unless this is true (that is, the value of the `if` expression is constant once `instantiatedExtensions` is created). Also consider checking for a plugin that overrides a method prior to returning a wrapped function. If no plugins override or filter a method the `baseService` version should be returned. In other words, a method should only be wrapped if there is at least one plugin that is interested in the method.

I expect many plugins to only override or filter one or two methods and this would mean that most of the methods never would check `instantiatedExtensions`.